fendermint/vm/interpreter/src/chain.rs

```diff
@@ -238,48 +246,31 @@ where
                         height: p.height as u64,
                         block_hash: p.block_hash,
                     };
-                    let msg = self
-                        .gateway_caller
-                        .commit_parent_finality_msg(finality.clone())?;
-                    let (state, ret) = self
-                        .inner
-                        .deliver(state, VerifiableMessage::NotVerify(msg))
-                        .await?;
-                    if ret.is_err() {
-                        return Err(anyhow!("cannot commit parent finality"));
-                    }
 
-                    // stash validator changes
+                    let (state, _, (prev_height, prev_finality)) = self
+                        .commit_finality(state, &provider, finality.clone())
+                        .await?;
+                    tracing::debug!("commit finality parsed: prev_height {prev_height}, prev_finality: {prev_finality:?}");
 
                     // error happens if we cannot get the validator set from ipc agent after retries
-                    let validator_changes = provider.validator_changes(p.height as u64).await?;
-                    let msg = self
-                        .gateway_caller
-                        .store_validator_changes_msg(validator_changes)?;
-                    let (state, ret) = self
-                        .inner
-                        .deliver(state, VerifiableMessage::NotVerify(msg))
+                    let validator_changes = provider
+                        .validator_changes_from(prev_height + 1, p.height as u64)
+                        .await?;
+                    let (state, _, _) = self
```

Comment:
The ignored variables are bothering me a little. Is one of them the `ret`? We should not ignore errors here. For now we should stick to returning an overall error if any of these fail.

Reply:
Right, I see that inside these methods they return errors. It's a bit confusing that everything is ignored here. 

Also, shouldn't we take at least the events from these and attach them to the final returned result?